libethereum/ChainParams.cpp

```diff
@@ -281,3 +287,55 @@ bytes ChainParams::genesisBlock() const
 	block.appendRaw(RLPEmptyList);
 	return block.out();
 }
+
+#include <libethashseal/GenesisInfo.h>
+js::mObject prepareFromGeneralConfig(js::mObject const& _config)
+{
+    assert(_config.count("params"));
+    assert(_config.count("state"));
+    assert(_config.count("genesis"));
+    js::mObject params = _config.at("params").get_obj();
+    assert(params.count("forkRules"));
+    string forkRules = params.at("forkRules").get_str();
+    Network rules;
+    if (forkRules == "Frontier")
+        rules = Network::FrontierTest;
+    else if (forkRules == "Homestead")
+        rules = Network::HomesteadTest;
+    else if (forkRules == "EIP150")
+        rules = Network::EIP150Test;
+    else if (forkRules == "EIP158")
```

Comment:
I'd prefer the general config it would be nice to refer to these with their semi-"official" names, such as SpuriousDragon and TangerineWhistle.

Reply:
Still we need docs on what those names mean exactly. What changes are applied.